File: javascript/fibonacci.js
function fibonacci(num) {
  // type your code here
  let fibonacciArray = [0, 1]

  for(i = 2; i <= num; i++){
    fibonacciArray[i] = fibonacciArray[i - 2] + fibonacciArray[i - 1]
  }

  return fibonacciArray[num]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));

  // my tests

  console.log("Expecting: 2")
  console.log("=>", fibonacci(3))

  console.log("Expecting: 5")
  console.log("=>", fibonacci(5))
}

module.exports = fibonacci;

// Please add your pseudocode to this file
  // given an array index, n, find the value in the fibonacci series that would be at that index

  // the value would be the sum of the two preceding values, so at index n-1 and n-2

  // fibonacci series starts with 0 and 1: [0, 1, 1, 2, 3, 5, 8]

  // initialize array [0, 1]

  // starting at index 2, for each element in the array, up to and including element n, set that index equal to the sum of the previous two elements

  // when nth element is reached, return value



// And a written explanation of your solution

  // this function starts the fibonacci series off with the first two numbers
  // then, given a specific index, n, the fibonacci array is calculated up to that element by adding the previous two elements together to supply the value of the next
  // the function then returns the value of the nth element

File: ruby/fibonacci.rb
def fibonacci(num)
  # type your code in here
  fibonacci_hash = [0, 1]

  index = 2

  while index <= num do
    fibonacci_hash[index] = fibonacci_hash[index - 1] + fibonacci_hash[index - 2]
    index += 1
  end

  fibonacci_hash[num]

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts "Expecting: 2"
  puts "=>", fibonacci(3)

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts "Expecting: 5"
  puts "=>", fibonacci(5)

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file

  # initialize an array with first two fibonacci elements: [0, 1]
  # loop through array, beginning at index 2, adding the previous two elements together to get the next value
  # stop when the nth element is reached
  # return value at index n

# And a written explanation of your solution
